cachito/web/models.py

```diff
@@ -647,8 +653,8 @@ class Request(db.Model):
     """A Cachito user request."""
 
     id = db.Column(db.Integer, primary_key=True)
-    repo = db.Column(db.String, nullable=False)
-    ref = db.Column(db.String, nullable=False)
+    repo = db.Column(db.String, nullable=False, index=True)
+    ref = db.Column(db.String, nullable=False, unique=True)
```

Comment:
I don't think we can use a unique constraint for `ref`. You can submit a request for the same `repo` and `ref` twice, which will result in two different rows in the db. Example:
```shell
for i in 1 2; do
  curl localhost:8080/api/v1/requests \
    -X POST \
    -H 'content-type: application/json' \
    -d '{
      "repo": "https://github.com/cachito-testing/cachito-pip-without-deps.git",
      "ref": "fb07c8492432631d557aba15e1715af9cef9c844",
      "pkg_managers": [
        "pip"
      ]
    }'
done

docker-compose exec db psql cachito -U cachito -c 'select * from request'
```
```
 id |                              repo                               |                   ref                    | user_id | submitted_by_id | request_state_id 
----+-----------------------------------------------------------------+------------------------------------------+---------+-----------------+------------------
  1 | https://github.com/cachito-testing/cachito-pip-without-deps.git | fb07c8492432631d557aba15e1715af9cef9c844 |         |                 |                8
  2 | https://github.com/cachito-testing/cachito-pip-without-deps.git | fb07c8492432631d557aba15e1715af9cef9c844 |         |                 |               14

```

Reply:
:sweat_smile:

Reply:
Updated.